src/Microsoft.ApplicationInsights.AspNet/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -48,6 +49,8 @@ public static void AddApplicationInsightsTelemetry(this IServiceCollection servi
             services.AddSingleton<ITelemetryInitializer, WebSessionTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, WebUserTelemetryInitializer>();
 
+            services.AddInstance<IConfiguration>(config);
```

Comment:
You can just pass this configuration to `ComponentVersionTelemetryInitializer` constructor on the line 44